blog/_posts/2020-05-16-next-few-years.adoc

```diff
@@ -0,0 +1,193 @@
+= rust-analyzer: Next Few Years
+:sectanchors:
+:experimental:
+:page-layout: post
+
+rust-analyzer is a new "IDE backend" for the https://www.rust-lang.org/[Rust] programming language.
+Support rust-analyzer on https://opencollective.com/rust-analyzer/[Open Collective].
+
+During the past several months, I've been swamped with in-the-trenches rust-analyzer work.
+Today, I spontaneously decided to take a step back and think about longer-term "road map" for rust-analyzer.
+
+What follows is my (https://github.com/matklad/[@matklad]) personal thoughts on the matter, they not necessary reflect the consensus view of ide or compiler teams :-)
+
+== Unexpected Success
+
+One of the most surprising aspects of rust-analyzer for me is how useful it already is.
+Today, I write Rust code enjoying fast code-completion, mostly correct go to definition and plethora of assists.
+Even syntax highlighting inside macros works!
+
+My original plan for rust-analyzer was to write a quick one-two year hack to demonstrate a proof-of-concept IDE support, something to strive for rather than a finished product.
+Obviously, we have massively overshoot this goal: people depend on rust-analyzer for productive Rust programming today.
+This creates its own opportunities and dangers, which inform this planing document.
+
+== rust-analyzer as the Official LSP Server for Rust
+
+People write a ton of Rust today, and they deserve at least a baseline IDE support.
+I think our immediate goal is to make rust-analyzer easier to use in its current state, effectively implementing https://github.com/rust-lang/rfcs/pull/2912[RFC2912].
+
+The amount of programming work on rust-analyzer side is relatively small here:
+we need to fix various protocol conformance issues,
+clean up various defaults to be less experimental,
+write documentation which doesn't require a lot of enthusiasm to understand, etc.
+The amount of org stuff is much bigger -- we need to package rust-analyzer with rustup, merge RLS and rust-analyzer VS Code extension, figure out repository structure, etc.
+
+Separately, I want to make sure that rust-analyzer is usable inside large non-Cargo based monorepos.
+We have some initial support for this already, but there's a bunch of details we need to iron out.
+
+== Dangers of Accidental Architecture
+
+The main danger I see is that rust-analyzer can ossify, as it is today.
```

Comment:
The main danger I see is that rust-analyzer can ossify, as it is today. => Suggestion: The main danger I see is that rust-analyzer can ossify in it's current/present state.
Reason: easier to understand the meaning (=> no strage thoughts like "is it ossified today already?")